var/spack/repos/builtin/packages/py-statsmodels/package.py

```diff
@@ -23,36 +26,50 @@ class PyStatsmodels(PythonPackage):
+    depends_on("python@:3.7", when="@0.10.1", type=("build", "link", "run"))
+    depends_on("python", type=("build", "link", "run"))
 
-    # according to https://www.statsmodels.org/dev/install.html earlier versions might work.
-    depends_on("py-setuptools", type="build")
-    depends_on("py-setuptools@59.2.0:", type="build", when="@0.13.5:")
+    # according to https://www.statsmodels.org/dev/install.html earlier versions
+    # might work.
+    depends_on("py-setuptools@59.2:", when="@0.13.3:", type="build")
+    depends_on("py-setuptools@0.6c5:", type="build")
 
-    # https://github.com/statsmodels/statsmodels/blob/01b19d7d111b29c183f620ff0a949ef6391ff8ee/pyproject.toml
-    depends_on("py-cython@0", type="build")
-    depends_on("py-cython@0.29.14:", type="build", when="@0.12.0:")
-    depends_on("py-cython@0.29.22:", type="build", when="@0.13.0:")
-    depends_on("py-cython@0.29.32:", type="build", when="@0.13.5:")
+    # pyproject.toml
+    depends_on("py-cython@0.29.26:2", when="@0.14:", type="build")
+    depends_on("py-cython@0.29.32:2", when="@0.13.5:0.13", type="build")
+    depends_on("py-cython@0.29.22:2", when="@0.13:", type="build")
+    depends_on("py-cython@0.29.14:2", when="@0.12:", type="build")
+    depends_on("py-cython@0.29:2", type="build")
+    depends_on("py-setuptools-scm+toml@7", when="@0.13.3:", type="build")
 
     # patsy@0.5.1 works around a Python change
     #    https://github.com/statsmodels/statsmodels/issues/5343 and
     #    https://github.com/pydata/patsy/pull/131
 
-    depends_on("py-numpy@1.11.0:", type=("build", "link", "run"), when="@0.10.1:")
-    depends_on("py-numpy@1.15.0:", type=("build", "link", "run"), when="@0.12.1:")
-    depends_on("py-numpy@1.17.0:", type=("build", "link", "run"), when="@0.13.0:")
-    depends_on("py-pandas@0.19:", type=("build", "run"), when="@0.10.1:")
-    depends_on("py-pandas@0.23:", type=("build", "run"), when="@0.12.0:")
-    depends_on("py-pandas@0.25:", type=("build", "run"), when="@0.13.0:")
-    depends_on("py-patsy@0.4.0:", type=("build", "run"), when="@0.10.1:")
-    depends_on("py-patsy@0.5.1:", type=("build", "run"), when="@0.12.0:")
-    depends_on("py-patsy@0.5.2:", type=("build", "run"), when="@0.13.0:")
-    depends_on("py-scipy@0.18:", type=("build", "run"), when="@0.10.1:")
-    depends_on("py-scipy@1.2:", type=("build", "run"), when="@0.12.0:")
-    depends_on("py-scipy@1.3:", type=("build", "run"), when="@0.13.0:")
-    depends_on("py-packaging@21.3:", type=("build", "run"), when="@0.13.2:")
+    # requirements.txt
+    depends_on("py-numpy@1.18:", when="@0.14:", type=("build", "link", "run"))
+    depends_on("py-numpy@1.17:", when="@0.13:", type=("build", "link", "run"))
+    depends_on("py-numpy@1.15:", when="@0.12.1:", type=("build", "link", "run"))
+    depends_on("py-numpy@1.11:", when="@0.10.1:", type=("build", "link", "run"))
+    depends_on("py-scipy@1.4:1.9.1,1.9.3:", when="@0.13.5:", type=("build", "run"))
```

Comment:
I usually do:
```suggestion
    depends_on("py-scipy@1.4:", when="@0.13.5:", type=("build", "run"))
    conflicts("^py-scipy@1.9.2")
```
for this kind of thing. If it's broken in newer releases, it's prob also broken in older releases too.